File: js/자료구조/stack.js
function Stack(array = []){
    this.array = array
}

// getBuffer() : 객체 내 데이터 셋 반환
Stack.prototype.getBuffer = function(){
    return [...this.array]
}
// isEmpty() : 객체 내 데이터가 있는지 없는지
Stack.prototype.isEmpty = function(){
    return this.array.length === 0
}

// push() : 데이터 추가
Stack.prototype.push = function(item){
    this.array.push(item)
    return
}
// pop() : 데이터 삭제 - 출력
Stack.prototype.pop = function(){
    return this.array.pop()
}
// size() : array의 사이즈
Stack.prototype.size = function(){
    return this.array.length
}
// peek() : array의 끝 데이터 반환
Stack.prototype.peek = function(){
    return this.array[this.array.length-1]
}
// indexOf() : item의 인덱스를 반환
Stack.prototype.indexOf = function(item){
    for(let i in this.array){
        if(this.array[i] === item) return i
    }
    return -1
}
// includes() : item을 포함하고 있는지
// Stack.prototype.includes = function(item){

// }


let stack = new Stack()
stack.push(123)
stack.push(123)
stack.push(123)
console.log(stack.isEmpty());
console.log(stack.getBuffer())
console.log(stack.size());

File: 알고리즘/구현/stack.py
class stack:
    def __init__(self,size):
        self.size = size
        self.items = []
    
    def isFull(self):
        if len(self.items) == self.size:
            return True
        return False
    
    def isEmpty(self):
        if len(self.items) == 0:
            return True
        return False
    
    def push(self, data):
        if not self.isFull():
            self.items.append(data)
            return f'{data} is pushed'
        return 'Stack is Full'
    
    def pop(self):
        if not self.isEmpty():
            item = self.items.pop()
            return f'{item} is poped'
        return 'Stack is Empty'
    
    def printStack(self):
        if not self.isEmpty():
            for item in self.items:
                print(item, end=' ')
            return
        return 'Stack is Empty'
    
stack1 = stack(10)

print(stack1.pop())
print(stack1.printStack())

print(stack1.push(5))
print(stack1.push(4))
print(stack1.push(3))
print(stack1.push(2))

for i in range(8):
    print(stack1.push(i))

        
print(stack1.pop())
stack1.printStack()